sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/noop/NoopSuite.scala

```diff
@@ -0,0 +1,78 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.execution.datasources.noop
+
+import org.apache.spark.sql.test.SharedSQLContext
+
+class NoopSuite extends SharedSQLContext {
+  import testImplicits._
+
+  test("materialisation of all rows") {
+    val numElems = 10
+    val accum = spark.sparkContext.longAccumulator
+    spark.range(numElems)
+      .map { x =>
+        accum.add(1)
+        x
+      }
+      .write
+      .format("noop")
+      .save()
+    assert(accum.value == numElems)
+  }
+
+  test("write partitioned data") {
+    val numElems = 100
+    val accum = spark.sparkContext.longAccumulator
+    spark.range(numElems)
+      .map { x =>
+        accum.add(1)
+        x
+      }
+      .select(('value mod 10) as "key", 'value)
+      .write
+      .format("noop")
+      .partitionBy("key")
```

Comment:
@cloud-fan, can you confirm if we support partitioned table in Data Source V2? If not supported, let's remove theses tests. Technically, this shouldn't be allowed for the current status.

Yes, if we can explicitly allow or disallow later, we can fix the tests but I think it's better to add the test later as a followup after we fix the codes about allowing and disallowing first. Currently, It gives an impression that we support partitioned tables.

Reply:
data source v2 doesn't support partition yet, let's remove the related tests.

Reply:
ok. I will remove the `write partitioned data` test but keep the `read partitioned data`.